openedx/core/djangoapps/user_authn/utils.py

```diff
@@ -57,28 +58,50 @@ def is_registration_api_v1(request):
     return 'v1' in request.get_full_path() and 'register' not in request.get_full_path()
 
 
-def generate_username_suggestions(username):
+def remove_special_characters_from_name(name):
+    return "".join(re.findall(r"[\w-]+", name))
+
+
+def generate_username_suggestions(name):
     """ Generate 3 available username suggestions """
+    username_suggestions = []
     max_length = USERNAME_MAX_LENGTH
-    short_username = username[:max_length - 6] if max_length is not None else username
-    short_username = short_username.replace('_', '').replace('-', '')
+    names = name.split(' ')
+    if names:
+        first_name = remove_special_characters_from_name(names[0].lower())
+        last_name = remove_special_characters_from_name(names[-1].lower())
+
+        if first_name != last_name:
+            # username combination of first and last name
+            suggestion = f'{first_name}{last_name}'[:max_length]
+            if not username_exists_or_retired(suggestion):
+                username_suggestions.append(suggestion)
 
-    username_suggestions = []
-    int_ranges = [
-        {'min': 0, 'max': 9},
-        {'min': 10, 'max': 99},
-        {'min': 100, 'max': 999},
-        {'min': 1000, 'max': 99999},
-    ]
-    for int_range in int_ranges:
-        for _ in range(10):
-            random_int = random.randint(int_range['min'], int_range['max'])
-            suggestion = f'{short_username}_{random_int}'
+            # username is combination of first letter of first name and last name
+            suggestion = f'{first_name[0]}-{last_name}'[:max_length]
```

Comment:
When a user enters spaces in the first name, it raises `IndexError: string index out of range`. Example input: `<space>rosy`